From 1782006a5cc6bdf61d09c312331b3e5f951ec605 Mon Sep 17 00:00:00 2001
From: Brian Bockelman <bbockelman@morgridge.org>
Date: Tue, 8 Feb 2022 07:44:57 -0600
Subject: [PATCH 13/17] XrdVomsMapfile: Add a comment on why we use ctime over
 mtime for detecting file changes.

(cherry picked from commit 5f0a255a58c01c49dc2aa7f67e760ca42deb7a21)
---
 src/XrdVoms/XrdVomsMapfile.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xrootd/src/XrdVoms/XrdVomsMapfile.cc b/xrootd/src/XrdVoms/XrdVomsMapfile.cc
index e53133433..30bdb88c9 100644
--- a/xrootd/src/XrdVoms/XrdVomsMapfile.cc
+++ b/xrootd/src/XrdVoms/XrdVomsMapfile.cc
@@ -400,6 +400,12 @@ XrdVomsMapfile::MaintenanceThread(void *myself_raw)
            myself->m_is_valid = false;
            continue;
        }
+       // Use ctime here as it is solely controlled by the OS (unlike mtime,
+       // which can be manipulated by userspace and potentially not change
+       // when updated - rsync, tar, and rpm, for example, all preserve mtime).
+       // ctime also will also be updated appropriately for overwrites/renames,
+       // allowing us to detect those changes as well.
+       //
        if ((myself->m_mapfile_ctime.tv_sec == statbuf.st_ctim.tv_sec) &&
            (myself->m_mapfile_ctime.tv_nsec == statbuf.st_ctim.tv_nsec))
        {
-- 
2.25.1

